spacy/tests/test_misc.py

```diff
@@ -138,3 +138,22 @@ def test_is_compatible_version(version, constraint, compatible):
 )
 def test_is_unconstrained_version(constraint, expected):
     assert util.is_unconstrained_version(constraint) is expected
+
+
+@pytest.mark.parametrize(
+    "dot_notation,expected",
+    [
+        (
+            {"token.pos": True, "token._.xyz": True},
+            {"token": {"pos": True, "_": {"xyz": True}}},
+        ),
+        (
+            {"training.batch_size": 128, "training.optimizer.learn_rate": 0.01},
+            {"training": {"batch_size": 128, "optimizer": {"learn_rate": 0.01}}},
+        ),
+    ],
+)
+def test_dot_to_dict(dot_notation, expected):
```

Comment:
This is a neat little function!

Reply:
Actually, I think this (or something like it) might be the missing piece for the oopen TODO in `debug model`: now that it takes the whole config and loads it, there needs to be a way for the user to specify which model to debug. So I thought on the CLI you could set `--section nlp.pipeline.tagger.model` and then it will select that component, if it exists.

Reply:
Done!